### DISCUSSION TITLE
Hardware recommendation

### DISCUSSION THREAD

**Discussion by vespaman** [blakeblackshear/frigate#13747] [General]
Hi all,
So, I am about to set up Frigate for the first time, and I have read the docs, and been trying to follow threads about hardware. But still I am a bit confused. 

I currently have about 6 cameras, 3k-4k resolution, and I want to setup Frigate on a new computer with low power consumption, and also want to be able to grow the camera system going forward.

Here's what I'm thinking

Hardware:
I guess my main contenders are either a N100 or an AMD Ryzen setup (e.g. ASUS PN5* series), where Ryzen seems to be consuming lower power, at least in idle, while also have more compute power if needed.
And then think I am heading for a Hailo-8L, which I think is now supported in Frigate(?).

There are new Ryzen series that has specifically AI/ML in mind, but I guess this is not supported by Frigate, or are they using openvino? Would it make sense to go for one of those instead?

To add to my confusion is the models
Is it possible to say that one is better over the others, and should this be part of the hardware decision?

Please give me advice and/or directions so I can have a piece of mind. :)

**Comment by NickM-27** [comment]
> And then think I am heading for a Hailo-8L, which I think is now supported in Frigate(?).

yes, in dev, will be there for the next release. If you go with intel then openvino can be used as a detector as well

> Is it possible to say that one is better over the others, and should this be part of the hardware decision?

yes, models can be affected by training methodology as well as the data it is trained on (and many other parameters as well). In general, I would not factor that in to the hardware decision.

**Reply by vespaman** [reply]
>yes, in dev, will be there for the next release.
As in 0.15.0? I.e. not 0.14.2 or something? Then I guess it means at least 6 months away, given the 0.14 was just released. Or maybe betas are available sooner.

> If you go with intel then openvino can be used as a detector as well
So, even if I go with a N100, you recon it would be somewhat useful also with 6 cameras?

OK, so regarding models, there's not one obvious leader with most goodies for the future among those that are available to this kind of hardware? That makes it easier.

What about AMD support then? Is this on the horizon, or best forget about it as of now?


Thanks for your input!

**Reply by NickM-27** [reply]
I have a WIP for AMD support but there are no models currently that perform well enough to be usable.

**Reply by vespaman** [reply]
So, I decided to go with a Beelink EQ13 (N100 for lower power), and just got the Hailo 8L as well (have not installed it yet).  I installed my favorite distro (gentoo) on it, with docker compose. The consumption without the Hailo idles at about 5W headless, one NIC in use, which is OK.

Now - I am thinking about next step.. I understand that next version will have the needed support for Hailo 8L, and if I'm not mistaken, there are no beta docker images for it(?). 
Given the super limited CPU, is there any point in even trying out current version of Frigate? Or should I simply wait? 
Or are there other ways? I.e. running some nightly build or something, without docker? Or is it a beast to do this, esp given I'm new to Frigate?

**Reply by NickM-27** [reply]
if you are not already, you should be running the model with your GPU using openvino

The next release is not in beta yet, there are dev images for testing but you should not expect stability since they are not beta yet https://github.com/users/blakeblackshear/packages/container/package/frigate

**Comment by vespaman** [comment]
Aha, OK, I'll test with openvino first. I got the feeling N100 would be too slow for anything like this, but at least I get too learn about the installation and setup this way.
Thanks!

**Reply by NickM-27** [reply]
We are aware of multiple users running the n100 with 10-16 cameras without issue

**Reply by vespaman** [reply]
Wow! That I did not expect. No hurry with the H8L in this case, I can really live with a bit slower detection time for a while.
I was envisioning perhaps following snails on the N100 without extra hardware. :)

**Reply by NickM-27** [reply]
Inference times on the GPU with openvino and with hailo8l should be similar for Frigate's usecase, I don't think you would notice much of a difference. They run the same model by default as well.

**Reply by vespaman** [reply]
Aha, so in fact, I might just forget about the h8l for this purpose? I guess I was thinking it would be needed,  because of the recommendation of using a Coral in the docs? Or is the Coral better suited than the H8L, in the Frigate use case?

**Reply by NickM-27** [reply]
so far the coral runs the default model faster than the hailo, but that could be due to differences in the model and the file formats, it remains to be seen. The hailo will definitely be able to run larger models than the coral can. 

The frigate docs recommend _a hardware detector_ which includes a coral, GPU, hailo, etc. They all perform very well for frigate's usecase

**Reply by vespaman** [reply]
Great, thanks for explaining! I'll pop the h8l in (making sure it will not consume an awful lot while in idle). Then I'll be ready for any upcoming enhancements.

**Reply by undrwater** [reply]
> Great, thanks for explaining! I'll pop the h8l in (making sure it will not consume an awful lot while in idle). Then I'll be ready for any upcoming enhancements.

It's been a while, but I'm following a similar trajectory. Did you ever get the H8L working, and if so, how is it going? 
Also, did you place it in the wifi/bt slot, or another.

Thanks!

**Reply by vespaman** [reply]
No, I never got around to use the H8L. I have been running my setup on the N100, which has worked pretty good, but I have intended to upgrade hardware at some point to be able to run better models. But I guess it runs good enough, as the project never seems to arrive to a hi priority.

**Reply by undrwater** [reply]
Thank you!
